### ISSUE TITLE
_deserialize_memoryview bug

### ISSUE THREAD

**Issue by adamklein** [dask/distributed#6542]
Small error in implementation.

Repro:

```python
from distributed.protocol import serialize, deserialize

x = memoryview(bytearray())
header, frames = serialize(x)
deserialize(header, frames)
```

Bug is here https://github.com/dask/distributed/blob/main/distributed/protocol/serialize.py#L786: 

Suggested fix:

```python
if out.nbytes > 0:
    out = out.cast(header["format"], header["shape"])
else:
    out = out.cast(header["format"])
```

Similar to #4594

cc @jakirkham

**Comment by jakirkham** [comment]
Yeah that seems like a reasonable solution. If you want to put it in a PR with a test, should be able to get it in before the upcoming release planned for this Friday ( https://github.com/dask/community/issues/252 )

Have some other local work to strip out empty frames during serialization. Though it's incomplete and probably won't get to it before the release unfortunately

**Comment by jakirkham** [comment]
@adamklein please let us know if `distributed` `main` works for you. Should have Conda nightlies shortly on the `dask` channel. Or can `pip install` directly from `main`

**Comment by jakirkham** [comment]
Should be included in 2022.6.1
